docs: invalid HammerLoader link reference

bougwal opened the pull request:
Fixes missing @ see to properly link to HammerLoader from Injection Token section 

<img width="1781" alt="Screenshot 2024-05-16 at 22 37 31" src="https://github.com/angular/angular/assets/61390636/a90596d3-16a4-4d32-af78-9716dd7dd5da">

JeanMeche commented on packages/platform-browser/src/dom/events/hammer_gestures.ts:
```diff
@@ -78,7 +78,7 @@ export const HAMMER_GESTURE_CONFIG = new InjectionToken<HammerGestureConfig>('Ha
 
 /**
- * Injection token used to provide a {@link HammerLoader} to Angular.
+ * Injection token used to provide a @see {@link HammerLoader} to Angular.
```
This creates a "pill" link. https://ng-dev-previews-fw--pr-angular-angular-55836-adev-prev-a1iofovk.web.app/api/platform-browser/HAMMER_LOADER 

I think we should fix that when generating the template over on the `dev-infra` repo.

bougwal replied:
yes right, it looked a bit strange from a UX standpoint (but functional)- Also adding that @ see would omit the text after the referenced link. In the deployed version the "to Angular" last part of the sentence is now omitted. Maybe a rephrase would be better now to put the link last and look at both issues ? wdyt @JeanMeche ?

JeanMeche replied:
Our template generator should be able to handle the link as is. I'd consider the curent behavior a bug. 

What we could do is removing the link a go with a separate pill : 

```
 * Injection token used to provide a HammerLoader to Angular.
 *
 * @see {@link HammerLoader}
```

bougwal replied:
<img width="1784" alt="Screenshot 2024-05-16 at 23 20 31" src="https://github.com/angular/angular/assets/61390636/0576054b-d974-4bde-95bb-df6879f63fd0">

JeanMeche commented:
Could you also make it a `docs: ...` commit message (no need for a scope), the `docs-infra` scope is meant for change on the adev app itself. Thank you.

bougwal committed:
docs: broken HammerLoader reference fix

*Fixing broken documentation reference to HammerLoader in HammerLoaderInjectionToken page

JeanMeche commented:
Note: We'll wait for angular/dev-infra#2069 to be merge to likely amend this PR.

JeanMeche commented:
I've the dev-infra update in #55888 but it turns out there is a bug in the typescript parsing, so we're stuck with an upstream issue. See

*  microsoft/TypeScript#58584

pkozlowski-opensource commented:
@JeanMeche @bencodezen is this one ready to merge? Looking at the comments history I'm getting conflicting messages: it is ready for merge but also there is a comment from @JeanMeche saying that this is blocked: https://github.com/angular/angular/pull/55836#issuecomment-2120455000

JeanMeche reviewed (changes requested):
We're waiting to land another fix

JeanMeche commented:
The TS 5.5 PR was merged, if you rebase on main the pills should now have the correct link.

thePunderWoman commented:
Caretaker note: this is docs only and is safe to merge without a presubmit

thePunderWoman commented:
This PR was merged into the repository by commit 03553c485d6f9ab0fbe6bfcb964b2076195efca2.

The changes were merged into the following branches: main, 18.1.x
